ImageLounge/src/DkGui/DkNoMacs.cpp

```diff
@@ -1272,23 +1272,47 @@ void DkNoMacs::openFile() {
 	if (fileNames.isEmpty())
 		return;
 
-	int count = getTabWidget()->getTabs().count();
+	int count = getTabWidget()->getTabs().count(); // Save current count of tabs for setting tab position later
 	if (getTabWidget()->getTabs().at(0)->getMode() == DkTabInfo::tab_empty)
 		count = 0; 
 		
-
+	QSet<QString> duplicates;
 	for (QString fileName : fileNames) {
-		qDebug() << "os filename: " << fileName;
-		getTabWidget()->loadFileToTab(fileName);
+		bool dup = false;
+
+		if (DkSettingsManager::param().global().checkOpenDuplicates) { // Should we check for duplicates?
+			for (auto tab : getTabWidget()->getTabs()) {
+				if (tab->getFilePath().compare(fileName) == 0) {
+					duplicates.insert(tab->getFilePath());
+					dup = true;
+					break;
+				}
+			}
+		}
+
+		if (!dup) {
+			qDebug() << "os filename: " << fileName;
+			getTabWidget()->loadFileToTab(fileName);
+		}
+	}
+	if (duplicates.count() > 0) { // Show messagebox if at least one duplicate was found
+		QMessageBox dupbox(this);

```

Comment:
I think a messagebox is too much here (note: we hardly ever use messageboxes), please use either:
- a DkMessageBox (for the user can permanently switch it off)
- `getTabWidget()->getViewPort()->getController()->setInfo(tr("some message"));`